src/main/java/com/minecolonies/colony/Field.java

```diff
@@ -0,0 +1,467 @@
+    {
+        this.fieldStage = fieldStage;
+    }
+
+    /**
+     * Checks if the field needs work (planting, hoeing)
+     * @return true if so.
+     */
+    public boolean needsWork()
+    {
+        return this.needsWork;
+    }
+
+    /**
+     * Sets that the field needs work
+     * @param needsWork true if work needed, false after completing the job.
+     */
+    public void setNeedsWork(boolean needsWork)
+    {
+        this.needsWork = needsWork;
+    }
+
+    /**
+     * Getter for MAX_RANGE.
+     * @return the max range.
+     */
+    public int getMaxRange()
+    {
+        return MAX_RANGE;
+    }
+
+    /**
+     * Getter of the seed of the field
+     * @return the ItemSeed
+     */
+    public Item getSeed()
+    {
+        //todo update seed @farmer
+        if(inventory.getStackInSlot(0) == null || !(inventory.getStackInSlot(0).getItem()  instanceof IPlantable))
+        {
+            return null;
+        }
+        return inventory.getStackInSlot(0).getItem();
+    }
+
+    /**
+     * Getter of the length in plus x direction.
+     * @return field length.
+     */
+    public int getLengthPlusX()
+    {
+        return lengthPlusX;
+    }
+
+    /**
+     * Getter of the with in plus z direction.
+     * @return field width.
+     */
+    public int getWidthPlusZ()
+    {
+        return widthPlusZ;
+    }
+
+    /**
+     * Getter of the length in minus x direction.
+     * @return field length.
+     */
+    public int getLengthMinusX()
+    {
+        return lengthMinusX;
+    }
+
+    /**
+     * Getter of the with in minus z direction.
+     * @return field width.
+     */
+    public int getWidthMinusZ()
+    {
+        return widthMinusZ;
+    }
+
+    /**
+     * Location getter.
+     * @return the location of the scarecrow of the field.
+     */
+    public BlockPos getLocation()
+    {
+        return this.location;
+    }
+
+    /**
+     * Return this citizens inventory.
+     *
+     * @return the inventory this citizen has.
+     */
+    @NotNull
+    public InventoryField getInventoryField()
+    {
+        return inventory;
+    }
+
+    /**
+     * Sets the inventory of the field.
+     * @param inventory the inventory to set.
+     */
+    public void setInventoryField(InventoryField inventory)
+    {
+        this.inventory = inventory;
+    }
+
+    @Override
+    public boolean canInteractWith(EntityPlayer playerIn)
+    {
+        return getColony().getPermissions().hasPermission(playerIn, Permissions.Action.ACCESS_HUTS);
+    }
+
+    @Override
+    public ItemStack transferStackInSlot(EntityPlayer playerIn, int slotIndex)
+    {
+        if (slotIndex == 0)
+        {
+            playerIn.inventory.addItemStackToInventory(inventory.getStackInSlot(0));
+            inventory.setInventorySlotContents(0, null);
+        }
+        else if(inventory.getStackInSlot(0) == null)
+        {
+            int playerIndex = slotIndex < 28 ? (slotIndex + 8) : (slotIndex - 28);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 28 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 8 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 28 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'playerIndex' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)